llama-index-integrations/vector_stores/llama-index-vector-stores-vearch/llama_index/vector_stores/vearch/base.py

```diff
@@ -0,0 +1,467 @@
+            )
+            res = query_result["hits"]["hits"]
+        else:
+            query_data = {
+                "vector": [
+                    {
+                        "field": "text_embedding",
+                        "feature": embed / np.linalg.norm(embed),
+                    }
+                ],
+                "fields": [],
+                "retrieval_param": {"metric_type": "InnerProduct", "efSearch": 64},
+                "topn": k,
+            }
+            query_result = self.vearch.search(query_data)
+            res = query_result[0]["result_items"]
+        nodes = []
+        similarities = []
+        ids = []
+        for item in res:
+            content = ""
+            meta_data = {}
+            node_id = ""
+            if self.flag:
+                score = item["_score"]
+                item = item["_source"]
+            for item_key in item:
+                if item_key == "text":
+                    content = item[item_key]
+                    continue
+                elif item_key == "_id":
+                    node_id = item[item_key]
+                    ids.append(node_id)
+                    continue
+                if self.flag != 1 and item_key == "score":
+                    score = item[item_key]
+                    continue
+                meta_data[item_key] = item[item_key]
+            similarities.append(score)
+            try:
+                node = metadata_dict_to_node(meta_data)
+                node.set_content(content)
+            except Exception:
+                metadata, node_info, relationships = legacy_metadata_dict_to_node(
+                    meta_data
+                )
+                node = TextNode(
+                    text=content,
+                    id_=node_id,
+                    metadata=metadata,
+                    start_char_idx=node_info.get("start", None),
+                    end_char_idx=node_info.get("end", None),
+                    relationships=relationships,
+                )
+            nodes.append(node)
+        return VectorStoreQueryResult(nodes=nodes, similarities=similarities, ids=ids)
+
+    def _delete(
+        self,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> Optional[bool]:
+        """
+        Delete the documents which have the specified ids.
+
+        Args:
+            ids: The ids of the embedding vectors.
+            **kwargs: Other keyword arguments that subclasses might use.
+
+        Returns:
+            Optional[bool]: True if deletion is successful.
+            False otherwise, None if not implemented.
+        """
+        ret: Optional[bool] = None
+        tmp_res = []
+        if ids is None or len(ids) == 0:
+            return ret
+        for _id in ids:
+            if self.flag:
+                ret = self.vearch.delete(self.using_db_name, self.using_table_name, _id)
+            else:
+                ret = self.vearch.del_doc(_id)
```

Comment:
Actually, the intention behind delete is to delete any node that has `ref_doc_id`/`_id` in its metadata under `doc_id` -- is this possible in vearch?

Reply:
vearch support deletion Specified id not all,thanks!

Reply:
So then this function isn't working properly -- it should be deleting every node that has `ref_doc_id` in its metadata under `doc_id` 👀

Reply:
Can you tell me what ref_doc_id is and the difference between doc_id? There may be some ambiguity in my understanding. vearch will delete all information corresponding to Doc_id according to DOC_ID

Reply:
_id in this code means doc_id which is user want to delete !

Reply:
So. When a document is broken into nodes, each node contains a reference to its parent doc `node.ref_doc_id`

When we call delete in a vector store, we want to delete all nodes that belong to a ref_doc_id

It's stored in the node metadata under a few names during the node_to_metadata_dict function, `doc_id` is one of those names I picked.

In any case, the idea is, we want to do a filtered delete here